students/tri_nguyen/lesson03/slicing.py

```diff
@@ -0,0 +1,144 @@
+# -------------------------------------------#
+# Student:   Tri Nguyen
+# Instructor: Natasha Aleksandrova
+# Date:  08-Feb-2018
+# -------------------------------------------#
+
+# Write some functions that take a sequence as an
+# argument, and return a copy of that sequence:
+# 1. With the first and the last item exchanged
+# 2. With every other item removed
+# 3. With the first 4 and the last 4 items removed
+# and then every other item in between
+# 4. With the elements reversed (just slicing)
+# 5. With the middle third, the last third, then
+# the first third in the new order
+
+# Part 1:
+
+def exchange_first_last(seq):
+    '''
+        this function takes a sequence (seq) and return
+        a copy with the first and the last item exchanged        
+    '''
+    if len(seq) <= 1:
+        exchanged_seq = 'Sequence has 0 or 1 value. Unable to compute.'
+    else:
+        exchanged_seq = seq[-1:] + seq[1:len(seq) - 1] + seq[:1]
+    return exchanged_seq
+    
+# Part 2:
+
+def remove_every_other(seq):
+    '''
+        this function takes a sequence (seq) and return
+        a copy with every other item removed    
+    '''
+    
+    container = []
+    
+    if len(seq) <= 1:
+        return 'Sequence has 0 or 1 value. Unable to compute.'
+    else:
+        for idx, item in enumerate(seq):
+            if not idx % 2:
+                container.append(str(item))
+         
+    return ' '.join(container)
+
+# With every other item removed can also be rewritten
+# as below re_every_other() function
+
+def re_every_other(seq):
+    '''
+        this function takes a sequence (seq) and return
+        a copy with the first and the last item exchanged        
+    '''
+    
+    if len(seq) <= 1:
+        return 'Sequence has 0 or 1 value. Unable to compute.'
+    else:
+        return seq[::2]
+
+
+# Part 3:
+
+def first_last_four(seq):
+    '''
+        this function takes a sequence (seq) and return
+        a copy with the first 4 and the last 4 items removed
+        and then every other item in between        
+    '''
+
+    if len(seq) <= 1:
+        return 'Sequence has 0 or 1 value. Unable to compute.'
+    else:
+        return seq[4:len(seq) - 4:2]
+
+# Part 4
+
+def reverse_seq(seq):
+    '''
+        this function takes a sequence (seq) and return
+        a copy with the elements reversed (just slicing)       
+    '''
+
+    if len(seq) <= 1:
+        return 'Sequence has 0 or 1 value. Unable to compute.'
+    else:
+        return seq[::-1]
+
+# Part 5
+
+def mid_last_first(seq):
+    '''
+        this function takes a sequence (seq) and return
+        a copy with the middle third, the last third, then
+        the first third in the new order
+    '''
+    
+    if len(seq) <= 1:
+        return 'Sequence has 0 or 1 value. Unable to compute.'
+    else:
+        first_third = seq[0:round(len(seq)/3)]
```

Comment:
for clarity extract `round(len(seq)/3)` into var or you can also use `i = len(seq)//3`